File: java/Motherboard.java
import java.util.ArrayList;
import java.util.List;

public class Motherboard extends Komponen {
    private String socketType;
    private String chipset;
    private int usb;
    private String formFactor;
    private List<Ram> rams;
    private CPU cpu;
    private CoolingSystem coolingSystem;
    private VGA vga;

    public Motherboard(String merk, String nama, String socketType, String chipset, int usb, String formFactor) {
        super(merk, nama);
        this.socketType = socketType;
        this.chipset = chipset;
        this.usb = usb;
        this.formFactor = formFactor;
        this.rams = new ArrayList<>();
        this.cpu = null;
        this.coolingSystem = null;
        this.vga = null;
    }

    public String getSocketType() {
        return socketType;
    }

    public void setSocketType(String socketType) {
        this.socketType = socketType;
    }

    public String getChipset() {
        return chipset;
    }

    public void setChipset(String chipset) {
        this.chipset = chipset;
    }

    public int getUsb() {
        return usb;
    }

    public void setUsb(int usb) {
        this.usb = usb;
    }

    public String getFormFactor() {
        return formFactor;
    }

    public void setFormFactor(String formFactor) {
        this.formFactor = formFactor;
    }

    public void addRam(Ram ram) {
        this.rams.add(ram);
    }

    public void setCpu(CPU cpu) {
        this.cpu = cpu;
    }

    public void setCoolingSystem(CoolingSystem coolingSystem) {
        this.coolingSystem = coolingSystem;
    }

    public void setVga(VGA vga) {
        this.vga = vga;
    }

    public String info() {
        StringBuilder result = new StringBuilder();
        result.append("Motherboard: ").append(getNama()).append(" ").append(chipset).append(" (Merk: ").append(getMerk()).append(")\n")
              .append("  Socket: ").append(socketType).append("\n")
              .append("  Form Factor: ").append(formFactor).append("\n")
              .append("  USB Ports: ").append(usb).append("\n")
              .append("  RAMs:\n");

        for (Ram ram : rams) {
            result.append("    - ").append(ram.info()).append("\n");
        }

        if (cpu != null) {
            result.append("  CPU: ").append(cpu.info()).append("\n");
        }

        if (coolingSystem != null) {
            result.append("  Cooling System: ").append(coolingSystem.info()).append("\n");
        }

        if (vga != null) {
            result.append("  VGA: ").append(vga.info()).append("\n");
        }

        return result.toString();
    }
}

File: Python/Motherboard.py
from Komponen import Komponen
from Ram import Ram
from CPU import CPU
from CoolingSystem import CoolingSystem
from VGA import VGA


class Motherboard(Komponen):
    def __init__(self, merk="", nama="", socket_type="", chipset="", usb=0, form_factor=""):
        super().__init__(merk, nama)
        self.socket_type = socket_type
        self.chipset = chipset
        self.usb = usb
        self.form_factor = form_factor
        self.rams = []
        self.cpu = None
        self.cooling_system = None
        self.vga = None

    def get_socket_type(self):
        return self.socket_type

    def get_chipset(self):
        return self.chipset

    def get_usb(self):
        return self.usb

    def get_form_factor(self):
        return self.form_factor

    def set_socket_type(self, socket_type):
        self.socket_type = socket_type

    def set_chipset(self, chipset):
        self.chipset = chipset

    def set_usb(self, usb):
        self.usb = usb

    def set_form_factor(self, form_factor):
        self.form_factor = form_factor

    def add_ram(self, ram):
        self.rams.append(ram)

    def set_cpu(self, cpu):
        self.cpu = cpu

    def set_cooling_system(self, cooling_system):
        self.cooling_system = cooling_system

    def set_vga(self, vga):
        self.vga = vga

def info(self):
    result = f"Motherboard: {self.get_nama()} {self.chipset} (Merk: {self.get_merk()})\n" \
             f"  Socket: {self.socket_type}\n" \
             f"  Form Factor: {self.form_factor}\n" \
             f"  USB Ports: {self.usb}\n"

    result += "  RAMs:\n"
    for ram in self.rams:
        result += f"    - {ram.info()}\n"

    if self.cpu:
        result += f"  CPU: {self.cpu.info()}\n"

    if self.cooling_system:
        result += f"  Cooling System: {self.cooling_system.info()}\n"

    if self.vga:
        result += f"  VGA: {self.vga.info()}\n"

    return result